test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayExtensionsTest.cs

```diff
@@ -0,0 +1,413 @@
+                .Verifiable();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model, viewEngine.Object);
+
+            // Act
+            var displayResult = helper.Display(expression: "SomeProperty", templateName: "SomeTemplate");
+
+            // Assert
+            viewEngine.Verify();
+        }
+
+        [Fact]
+        public void Display_UsesTemplateNameAndAdditionalViewData()
+        {
+            // Arrange
+            var model = new SomeModel { SomeProperty = "ModelValue" };
+            var view = new Mock<IView>();
+            view.Setup(v => v.RenderAsync(It.IsAny<ViewContext>()))
+                .Callback(async (ViewContext v) =>
+                {
+                    await v.Writer.WriteAsync(v.ViewData["SomeProperty"].ToString());
+                })
+                .Returns(Task.FromResult(0));
+            var viewEngine = new Mock<ICompositeViewEngine>(MockBehavior.Strict);
+            viewEngine
+                .Setup(v => v.GetView(/*executingFilePath*/ null, It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound(string.Empty, Enumerable.Empty<string>()));
+            viewEngine
+                .Setup(v => v.FindView(It.IsAny<ActionContext>(), "DisplayTemplates/SomeTemplate", /*isMainPage*/ false))
+                .Returns(ViewEngineResult.Found("SomeView", view.Object));
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model, viewEngine.Object);
+
+            // Act
+            var displayResult = helper.Display(
+                expression: "SomeProperty",
+                templateName: "SomeTemplate",
+                additionalViewData: new { SomeProperty = "ViewDataValue" });
+
+            // Assert
+            Assert.Equal("ViewDataValue", HtmlContentUtilities.HtmlContentToString(displayResult));
+        }
+
+        [Fact]
+        public void Display_UsesTemplateNameAndHtmlFieldName()
+        {
+            // Arrange
+            var model = new SomeModel { SomeProperty = "ModelValue" };
+            var view = new Mock<IView>();
+            view.Setup(v => v.RenderAsync(It.IsAny<ViewContext>()))
+                .Callback(async (ViewContext v) =>
+                {
+                    await v.Writer.WriteAsync(v.ViewData.TemplateInfo.HtmlFieldPrefix);
+                })
+                .Returns(Task.FromResult(0));
+            var viewEngine = new Mock<ICompositeViewEngine>(MockBehavior.Strict);
+            viewEngine
+                .Setup(v => v.GetView(/*executingFilePath*/ null, It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound(string.Empty, Enumerable.Empty<string>()));
+            viewEngine
+                .Setup(v => v.FindView(It.IsAny<ActionContext>(), "DisplayTemplates/SomeTemplate", /*isMainPage*/ false))
+                .Returns(ViewEngineResult.Found("SomeView", view.Object))
+                .Verifiable();
```

Comment:
Verification doesn't matter much in this test since this setup is the only way the `view` could be used. Same for `DisplayFor` version.